File: api/src/main/java/com/api/app/entity/CodeDetail.java
package com.api.app.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.type.Alias;

import java.io.Serializable;

/**
 * @author system
 * @version 1.0
 * @since 2025-10-27
 */
@Alias("CodeDetail")
@Getter
@Setter
public class CodeDetail implements Serializable {
    private static final long serialVersionUID = 8901234567890123456L;

    @Schema(description = "그룹코드")
    private String groupCode;

    @Schema(description = "코드")
    private String code;

    @Schema(description = "코드명")
    private String codeName;

    @Schema(description = "참조값1")
    private String referenceValue1;

    @Schema(description = "참조값2")
    private String referenceValue2;

    @Schema(description = "정렬순서")
    private Long displaySequence;
}

File: api/monolith/src/main/kotlin/com/api/app/entity/CodeDetail.kt
package com.api.app.entity

import jakarta.persistence.*
import java.io.Serializable

@Embeddable
data class CodeDetailId(
    @Column(name = "GROUP_CODE", length = 15) val groupCode: String = "",
    @Column(name = "CODE", length = 3) val code: String = ""
) : Serializable

@Entity
@Table(name = "CODE_DETAIL")
class CodeDetail : SystemEntity() {

    @EmbeddedId
    var id: CodeDetailId = CodeDetailId()

    @Column(name = "CODE_NAME", nullable = false, length = 100)
    var codeName: String = ""

    @Column(name = "REFERENCE_VALUE_1", length = 255)
    var referenceValue1: String? = null

    @Column(name = "REFERENCE_VALUE_2", length = 255)
    var referenceValue2: String? = null

    @Column(name = "DISPLAY_SEQUENCE", columnDefinition = "numeric")
    var displaySequence: Long = 0

    // Convenience accessors
    val groupCode: String get() = id.groupCode
    val code: String get() = id.code
}
